main.go

```diff
@@ -65,6 +65,7 @@ func main() {
 
 	r.HandleFunc("/tasks/{key}", setup(controllers.TaskByKey))
 	r.HandleFunc("/tasks", setup(controllers.Tasks))
+	r.HandleFunc("/task/code", setup(controllers.CreateCodeTask))
```

Comment:
I already changed most of this. Waiting for your PR to get on the master so i can rebase and push